interop-testing/src/test/java/io/grpc/testing/integration/TlsTest.java

```diff
@@ -0,0 +1,253 @@
+    int port = TestUtils.pickUnusedPort();
+
+    // Create & start a server.
+    File serverCertFile = TestUtils.loadCert("localhost_server.pem");
+    File serverPrivateKeyFile = TestUtils.loadCert("localhost_server.key");
+    X509Certificate[] serverTrustedCaCerts = {
+      TestUtils.loadX509Cert("ca.pem")
+    };
+    Server server = serverBuilder(port, serverCertFile, serverPrivateKeyFile, serverTrustedCaCerts)
+        .addService(EchoServiceGrpc.bindService(new DummyEchoRpcService()))
+        .build()
+        .start();
+
+    try {
+      // Create a client.
+      File clientCertFile = TestUtils.loadCert("client.pem");
+      File clientPrivateKeyFile = TestUtils.loadCert("client.key");
+      X509Certificate[] clientTrustedCaCerts = {
+        TestUtils.loadX509Cert("ca.pem")
+      };
+      ManagedChannel channel = clientChannel("localhost", port, clientCertFile,
+                                             clientPrivateKeyFile, clientTrustedCaCerts);
+      EchoServiceBlockingStub client = EchoServiceGrpc.newBlockingStub(channel);
+
+      // Send an actual request, via the full GRPC & network stack, and check that a proper
+      // response comes back.
+      EchoRequest request = EchoRequest.newBuilder()
+          .setText("dummy text")
+          .build();
+      EchoResponse response = client.echo(request);
+      assertEquals("Request said: dummy text", response.getText());
+    } finally {
+      server.shutdown();
+    }
+  }
+
+
+  /**
+   * Tests that a server configured to require client authentication refuses to accept connections
+   * from a client that has an untrusted certificate.
+   */
+  @Test
+  public void serverRejectsUntrustedClientCert() throws Exception {
+    int port = TestUtils.pickUnusedPort();
+
+    // Create & start a server. It requires client authentication and trusts only the test CA.
+    File serverCertFile = TestUtils.loadCert("localhost_server.pem");
+    File serverPrivateKeyFile = TestUtils.loadCert("localhost_server.key");
+    X509Certificate[] serverTrustedCaCerts = {
+      TestUtils.loadX509Cert("ca.pem")
+    };
+    Server server = serverBuilder(port, serverCertFile, serverPrivateKeyFile, serverTrustedCaCerts)
+        .addService(EchoServiceGrpc.bindService(new DummyEchoRpcService()))
+        .build()
+        .start();
+
+    try {
+      // Create a client. Its credentials come from a CA that the server does not trust. The client
+      // trusts both test CAs, so we can be sure that the handshake failure is due to the server
+      // rejecting the client's cert, not the client rejecting the server's cert.
+      File clientCertFile = TestUtils.loadCert("badclient.pem");
+      File clientPrivateKeyFile = TestUtils.loadCert("badclient.key");
+      X509Certificate[] clientTrustedCaCerts = {
+        TestUtils.loadX509Cert("ca.pem"),
+        TestUtils.loadX509Cert("badclient.pem")  // Cert is self-signed, and so is its own issuer.
```

Comment:
Agreed that it *shouldn't* be necessary. The purpose of including this cert here was to avoid the following situation:

* For some reason the client performs validation that it needn't perform.
* The server fails to validate the client cert and just blindly accepts it. (This is [already happening](https://github.com/grpc/grpc-java/issues/1325).)
* The test passes, and so we have no way to know that all that is going wrong.

Just clarifying the rationale -- whichever way you think best is fine with me.